### PR TITLE
Add Search Expression support

### PR DESCRIPTION
**Pull request by l2yao**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-experiments/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-experiments/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-experiments/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-experiments/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/smexperiments/search_expression.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+@unique
+class Operator(Enum):
+    EQUALS = "Equals"
+    NOT_EQUALS = "NotEquals"
+    GREATER_THAN = "GreaterThan"
+    GREATER_THAN_OR_EQUAL = "GreaterThanOrEqualTo"
+    LESS_THAN = "LessThan"
+    LESS_THAN_OR_EQUAL = "LessThanOrEqualTo"
+    CONTAINS = "Contains"
```
Comment (danabens) [inline src/smexperiments/search_expression.py:26]: should the "In" operator also be in this list?

File: tests/integ/test_search_expression.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from smexperiments.experiment import Experiment
+import pytest
+
+
+def test_search(sagemaker_boto_client):
+    experiment_names_searched = []
+    search_filter = Filter(name="ExperimentName", operator=Operator.CONTAINS, value="smexperiments-integ-")
+    search_expression = SearchExpression(filters=[search_filter])
+    for s in Experiment.search(
+        search_expression=search_expression.to_boto(), max_results=10, sagemaker_boto_client=sagemaker_boto_client
```
Comment (danabens) [inline tests/integ/test_search_expression.py:23]: can we avoid calling code using "to_boto"?  Seems like the usage would be cleaner without it.

File: src/smexperiments/search_expression.py (hunk 1 above)
Comment (l2yao) [reply src/smexperiments/search_expression.py:26]: Operator list is taken from boto spec: https://boto3.amazonaws.com/v1/documentation/api/1.12.8/reference/services/sagemaker.html#SageMaker.Client.search

I only listed all available options from there.

File: tests/integ/test_search_expression.py (hunk 2 above)
Comment (l2yao) [reply tests/integ/test_search_expression.py:23]: agree and removed the need for to_boto here

File: src/smexperiments/search_expression.py (hunk 1 above)
Comment (wangweixun) [reply src/smexperiments/search_expression.py:26]: IN will be released today.

**Pull request merged**